ui/app/css/utilities/_colors.scss

```diff
@@ -0,0 +1,65 @@
+@use "sass:math";
+@use "sass:map";
+
+/**
+ * Calculate the luminance for a color.
+ * See https://www.w3.org/TR/WCAG20-TECHS/G17.html#G17-tests
+ */
+@function luminance($color) {
```

Comment:
Wow, this is a pretty complex function!

I did some Google-ing about this and stumbled across this interesting post: https://css-tricks.com/programming-sass-to-create-accessible-color-combinations/
They reported that when using this method:

>Unfortunately, calculating only a handful of color contrast combinations increased Sass build times exponentially

But they managed to solve it more performantly in the end using a lookup table.

Not saying we should optimize this now before it becomes a problem, but... this is something to consider if we notice a slowdown.

Reply:
i'm going to leave a note in the code about this. I *think* that the expensive part of the operation was the [custom `pow` function](https://css-tricks.com/snippets/sass/power-function/#positive-or-negative-exponents) that didn't exist in older versions of sass. To get around this some implementations implemented functions that do many loops. Since then, `math.pow` is a language level feature (https://github.com/sass/dart-sass/releases/tag/1.25.0). I tried to find any mentions of it's performance, but came up empty.

Reply:
Ah neat, TIL.